### DISCUSSION TITLE
Rest Api error message

### DISCUSSION THREAD

**Discussion by McMike4** [curl/curl#14225] [Q&A] [answered]
![error](https://github.com/user-attachments/assets/302582c9-a9d4-499b-a6c1-6c254388a247)

Please could someone help me to resolve this?

**Comment by icing** [comment, marked as answer]
try the user in double quotes, like:

```
curl --user "restapi3:..." -i http://...
```

**Reply by McMike4** [reply]
![error2](https://github.com/user-attachments/assets/5aba33d7-99ee-4651-a84f-bd644fe17d95)

I still got this error message

**Comment by icing** [comment]
@McMike4 the command line is chopped up into pieces where a space character is found. `--user` takes the next argument. That is `restapi2:9Rr4` in your example. But you want it to be `restapi2:9Rr4 AhOD nhUQ Slf5 jgug BS63`. Do make the command line do this, you put all these inside double quotes, as in `"restapi2:9Rr4 AhOD nhUQ Slf5 jgug BS63"`, like a citation. That prevents the windows shell to chop it into pieces.

**Comment by McMike4** [comment]
![error 3](https://github.com/user-attachments/assets/0cb6714f-6259-4d7a-9101-22a1cc9603f8)

Now I've got this error message coming out. Do you know the reason why?

**Reply by icing** [reply]
remove the '-X'.

**Comment by McMike4** [comment]
you mean remove the '-I'?

**Comment by McMike4** [comment]
![error 4](https://github.com/user-attachments/assets/d33f6518-be79-4805-9d1f-28b52dbc80af)

This is what I've got now

**Reply by jzakrzewski** [reply]
Not `curl GET`
Not `curl -X`
Just `curl`

To be exact, you want exactly this command line:
`curl --user "restapi2:9Rr4 4hOD nhUQ Slf5 jgug BS63" "http://rest-api-2.local.crt/wp-json/wp/v2/posts?status=draft"`

If you still get "Could not resolve host: rest-api-2.local.crt" message, then your environment is wrong.

**Reply by McMike4** [reply]
based on you command line I've got this message in return:
curl: (6) Could not resolve host: http

**Reply by jzakrzewski** [reply]
Just corrected a typo - missed a colon after http

**Reply by McMike4** [reply]
After typing the command I've got this error message that you can see in the screenshot attached. Do you know the reason?
Thanks in advance
![404 not found](https://github.com/user-attachments/assets/d620affa-ff22-437d-a42c-498b46951a77)

**Reply by dfandrich** [reply]
The meaning of the HTTP/1.1 404 Not Found error is that the origin server did not find a current representation for the target resource or is not willing to disclose that one exists. i.e. the URL is bad.

These questions don't have anything to do with curl. You should fist read up on the basics of HTTP and the documentation specifically on the HTTP API you're trying to access in order to make sense of what you're seeing.

**Comment by McMike4** [comment]
![error 5](https://github.com/user-attachments/assets/9e5688c5-ff9a-4b15-ab19-6e03a0a56101)

I've got this now after correcting the mistake

**Reply by jzakrzewski** [reply]
So it's working now. You get an empty JSON array back.

**Reply by McMike4** [reply]
why do I have an invalid ID on this one?
![error 6](https://github.com/user-attachments/assets/7ef74b29-6ad8-439a-9925-7c0c4ece7e2c)

**Reply by dfandrich** [reply]
That's a response from the application that curl is returning to you. Presumably, the application wants different data in order to respond differently, but that's not an issue with curl.

**Reply by McMike4** [reply]
I just want to post this content on my local site as a test but it just returns this error message.. Do you know how to fix this?

**Reply by dfandrich** [reply]
I know nothing about your web application. But I'm guessing you need to supply it a valid post ID.

**Comment by McMike4** [comment]
![error 7](https://github.com/user-attachments/assets/5f4dfe97-fb54-4593-bbb7-97047660019a)

Do you know what's the problem here? I can't see any change expected in the website.

**Reply by dfandrich** [reply]
If you're not getting the data you expect, you'll need to bring it up with the server operator. I see no curl issue here.

**Reply by McMike4** [reply]
I am testing the rest api on Local Flywheel. Have you already used it? I am trying to find out but still doesn't work..

**Reply by dfandrich** [reply]
You'll have better luck finding a forum where people know what Local Flywheel is.

**Reply by McMike4** [reply]
Hello, I've got the same issue while I am using Wampserver. The changes expected on the website page don't show up. It just remains the same page.
![spc](https://github.com/user-attachments/assets/03cf49de-ba7f-4799-982e-dec34cb6b553)

**Comment by McMike4** [comment]
Can we use email addresses as username in the command in the screenshot?
![cmd](https://github.com/user-attachments/assets/2646d028-9279-4a1b-8eb9-f38b7393fc9c)

**Reply by McMike4** [reply]
As you can see in this screenshot there is something wrong indeed
![returned message](https://github.com/user-attachments/assets/736d8b41-7992-4809-b8a7-9ba5538fbc66)

**Reply by dfandrich** [reply]
Curl can use just about anything you want as username. Your command as given is going to use HTTP Basic authentication, which isn't how most HTTP APIs are authenticated. You will need to read up the documentation on the API you're using to find out how they want you to authenticate.

Note also that if you ran the command you give in the top screenshot, your username *and password* would have been sent to the server unencrypted and therefore exposed to anyone with access to the network between you and the server.
